### ISSUE TITLE
Log name(Cached_name) Mismatch or wrong

### ISSUE THREAD

**Issue by umaraliyar** [creacore-team/cordova-plugin-calllog#10]
Hi,
Thanks for the plugin & maintenance.  Im using Cordova & Javascript to retrieve log information after passing request permission. I am getting wrong "name" for the right numbers from data return. Esp., number starts with 63... others are ok with proper name. As native Android suggest, should I go search in contact to get the right name? if so im not finding a document to search with this plugin. Guide me.

**Comment by Bruno361** [comment]
Hi,
The cached_name can indeed be wrong, that's why the plugin performs a contact lookup on the phone number, which you can find in the "name" value of the logs.
If that does not suit your needs, you should indeed perform a separate contact search, which this plugin cannot provide.

**Comment by umaraliyar** [comment]
Thank you for your comment. I am using name value from the log and I’m getting wrong name.

**Comment by Bruno361** [comment]
What kind of wrong name do you get ? Another one from your contact list ?
What do the two phone numbers look like, then ? Do you think the contact lookup gets mixed up somehow ?

**Comment by umaraliyar** [comment]
Yes another name from contact as mentioned when it starts from 63... I have gone through the possibilities like checking if I saved the number on both the contacts. I even deleted the contact and tried.  Later it took another name say "Name A" was showing before after deleting the contact is showing "Name B". I tried with different phone numbers starts with 6... All shows the same type of issues.

**Comment by VoltX** [comment]
Hi @umaraliyar, we fixed the cached number in our last 1.3.2 version.

Does it solve your problem ?

**Comment by f18nfz** [comment]
Did your problem get resolved?

 I noticed that if I dial (a random unsaved) number out, it always shows a random contact name (cached_name) with it incorrectly. I keep having this issue on whichever device I test on. I'm just using the example code... Is it just me? Any ideas?

FYI: If I delete the contact that is incorrectly shown against the number I dialed, it'll just give another random contact name as the (cached_name) name against the dialed number. >_<
